scalding-core/src/main/scala/com/twitter/scalding/typed/WithReducers.scala

```diff
@@ -24,6 +24,18 @@ trait HasReducers {
 }
 
 /**
+ * used for types that must know how many reducers they need
+ * e.g. Sketched
+ */
+trait MustHasReducers {
+  def reducers: Option[Int]
```

Comment:
Let's not do the runtime exception. Instead can we do this:

```scala
trait MustHaveReducers extends HasReducers {
  def reducers: Some[Int]
}
```
This should show that reducers is a Some[T] which is a subclass of Option[T]. That should allow us at compile time to verify the property you are checking at runtime below.